memory-management/src/dynamic_layout.rs

```diff
@@ -0,0 +1,195 @@
+//! Dynamic memory layout containers
+
+/// An array of equally spaced values
+#[derive(Clone)]
+#[repr(C)]
+pub struct DynamicLayoutArray<'a, T> {
+    _element_type: std::marker::PhantomData<&'a T>,
+    element_count: u32,
+    values_stride: u32,
+    values_offset: u32,
+}
+
+impl<'a, T> DynamicLayoutArray<'a, T> {
+    pub fn new_strided(start_offset: usize, element_count: usize, values_stride: usize) -> Self {
```

Comment:
I think the constructor should be fallible and fail if start_offset and values_stride don't respect alignment constraints